Core/Python/add_device_to_static_group.py

```diff
@@ -57,32 +182,56 @@ def add_device_to_static_group(ome_ip_address, ome_username, ome_password, group
                                      headers=headers)
         if session_info.status_code == 201:
             headers['X-Auth-Token'] = session_info.headers['X-Auth-Token']
-            response = requests.get(group_url, headers=headers, verify=False)
-            if response.status_code == 200:
-                json_data = response.json()
-                if json_data['@odata.count'] > 0:
-                    # Technically there should be only one result in the filter
-                    group_id = json_data['value'][0]['Id']
-                    group_payload = {"GroupModel": {
-                        "Name": group_name,
-                        "Description": "",
-                        "MembershipTypeId": 12,
-                        "ParentId": int(group_id)}
-                    }
-                    create_url = 'https://%s/api/GroupService/Actions/GroupService.CreateGroup' % (ome_ip_address)
-                    create_resp = requests.post(create_url, headers=headers,
-                                                verify=False,
-                                                data=json.dumps(group_payload))
-                    if create_resp.status_code == 200:
-                        print("New group created : ID =", create_resp.text)
-                    elif create_resp.status_code == 400:
-                        print("Failed group creation ...See error info below")
-                        print(json.dumps(create_resp.json(), indent=4,
-                                         sort_keys=False))
-            else:
-                print("Unable to retrieve group list from %s" % ome_ip_address)
-        else:
-            print("Unable to create a session with appliance %s" % ome_ip_address)
+
+            group_id = get_group_id_by_name(ome_ip_address, group_name, headers)
+
+            device_ids = []
+            if device_names:
+                for device in device_names:
+                    device_id = get_device_id_by_name(ome_ip_address, device, headers)
+                    if device_id != 0:
+                        device_ids.append(device_id)
+            elif device_tags:
+                device_list = get_device_list(ome_ip_address, headers)
+
+                # Create id - service tag index to avoid O(n) lookups on each search
```

Comment:
This could have been done in the method to get device list - the method could have returned a dictionary of service tags mapped to id unless I am missing something and the list is used elsewhere in the code

Reply:
Updated